low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -32,18 +44,125 @@ class ServerMap(ResourceMap):
+        else:
+            fru = None
+            fru_found = False
+            for node in nodes:
+                fru, _ = self.get_node_details(node)
+                if self.server_map.get(fru):
+                    fru_found = True
+                    info = self.server_map[fru]()
+                    break
+        if not fru_found:
+            raise ResourceMapError(
+                errno.EINVAL,
+                f"Health provider doesn't have support for'{rpath}'.")
+        return info
+
+    def get_mem_info(self):
+        """Update and return memory info in specific format"""
+
+        from framework.utils.store_factory import file_store
+        node_id = Conf.get(GLOBAL_CONF, NODE_ID_KEY,'SN01')
+        self.store = file_store
+        cache_dir_path = os.path.join(DATA_PATH, 'server')
+        self.MEM_FAULT_SENSOR_DATA = os.path.join(cache_dir_path,
+                                             f'MEM_FAULT_SENSOR_DATA_{node_id}')
+        data = []
+        status = "OK"
+        description = "Host memory is in good health"
+        utility_instance = ToolFactory().get_instance('procfs')
+        mem_path = utility_instance.get_proc_memory('meminfo')
+        if mem_path.is_file():
+            meminfo_dict = self.read_mem_file(mem_path)
+            total_mem = meminfo_dict['MemTotal:']
+            path_exists, _ = self.store.exists(self.MEM_FAULT_SENSOR_DATA)
+            if path_exists:
+                self.get_stored_mem_info()
+                if self.fault_alert_state == "fault":
+                    status = "FAULT"
+                    description  = \
+                    "Total available main memory value decreased"\
+                    f"from {self.prev_mem} kB to {total_mem} kB"
+            memory_dict = self.prepare_mem_json(status,
+                                                description)
+            data.append(memory_dict)
+        else:
+            raise HealthProviderError(
+                errno.EINVAL,
+                f"{mem_path}: file does not exist")
+        return data
+
+    def read_mem_file(self, mem_path):
+        """ Get Memory information from /proc/meminfo."""
+        meminfo_dict = {}
+        self.mem_path_file = mem_path.read_text()
+        # populate all the data from /proc/meminfo
+        split_strs = [s.split(maxsplit=1) for s in self.mem_path_file.splitlines()]
+        meminfo_dict = dict(split_strs)
+        return meminfo_dict
+
+    def get_stored_mem_info(self):
+        """ Get the memory info from filestore """
+        mem_data = (self.store.get(self.MEM_FAULT_SENSOR_DATA)).split(":")
+        self.prev_mem = mem_data[0].strip()
+        self.fault_alert_state = mem_data[1].strip()
+
+    def prepare_mem_json(self, status, description):
+        """Update and return memory information dict """
+        values = dict(psutil.virtual_memory()._asdict())
+        total_memory = {}
+        for key, value in values.items():
```

Comment:
Is value unused? If only key has scope in this iteration, then `for key in values' would be sufficient.

Reply:
done.